extensions/amp-video/0.1/flexible-bitrate.js

```diff
@@ -102,22 +105,29 @@ export class BitrateManager {
     if (video.changedSources) {
       return;
     }
-    onNontrivialWait(video, () => {
-      const current = currentSource(video);
-      const newBitrate = current.bitrate_ - 1;
-      if (newBitrate >= this.acceptableBitrate_) {
-        return;
-      }
-      this.acceptableBitrate_ = newBitrate;
-      this.switchToLowerBitrate_(video, current.bitrate_);
-      this.updateOtherManagedAndPausedVideos_();
-    });
+    onNontrivialWait(video, () => this.downgradeVideo_(video));
+    listen(video, 'downgrade', () => this.downgradeVideo_(video));
     video.changedSources = () => {
       this.sortSources_(video);
     };
     this.videos_.push(DomBasedWeakRef.make(this.win, video));
   }
 
+  /**
+   * Callback to try downgrade a video when it doesn't load properly.
```

Comment:
Nits on the comment:
- "Callback": it doesn't have to be a callback it can be called anytime
- "when it doesn't load properly": you don't have to guess when a method will be called, this will very likely become obsolete/inaccurate with time, you can instead focus on what the method does

Suggestion: "Downgrade a video quality by selecting a lower bitrate source if available, then downgrade the other registered videos"